cluster/gce/gci/configure-helper.sh

```diff
@@ -1781,7 +1781,27 @@ function start-kube-proxy {
 # $5: pod name, which should be either etcd or etcd-events
 function prepare-etcd-manifest {
   local host_name=${ETCD_HOSTNAME:-$(hostname -s)}
-  local -r host_ip=$(python3 -c "import socket;print(socket.gethostbyname(\"${host_name}\"))")
+
+  read -r -d '' resolve_host_script_py << EOM
+import socket
+import time
+import sys
+
+timeout_sec=300
+
+def resolve(host):
+  for attempt in range(timeout_sec):
+    try:
+      print(socket.gethostbyname(host))
+      break
+    except Exception as e:
+      sys.stderr.write("error: resolving host %s to IP failed: %s\n" % (host, e))
+      time.sleep(1)
+      continue
+EOM
+
+  local nl=$'\n'
+  local -r host_ip=$(python3 -c "${resolve_host_script_py};${nl}${nl}resolve(\"${host_name}\")")
```

Comment:
Why not doing what we did internally instead (i.e. something like:

host_ip=$(ETCD_IP:-127.0.0.1)

?
[and allow setting ETCD_IP explicitly if needed]

Reply:
I assume it would be a breaking change to all users that do set ETCD_HOSTNAME explicitly already
and depend on auto-resolution of this variable to IP address. 

But I don't know all the usage modes, so tried to make as safe change as possible, that can be even backported. 

If we want to go that path, I would consider: 
```
host_ip=$(ETCD_IP:hostname -i)
```
for symmetry with `hostname -s`